pkg/core/resources/apis/mesh/dataplane_validator.go

```diff
@@ -134,9 +130,21 @@ func validateGateway(gateway *mesh_proto.Dataplane_Networking_Gateway) validator
 	if _, exist := gateway.Tags[mesh_proto.ServiceTag]; !exist {
 		result.AddViolationAt(validators.RootedAt("tags").Key(mesh_proto.ServiceTag), `tag has to exist`)
 	}
-	for name, value := range gateway.Tags {
+	result.Add(validateTags(gateway.Tags))
+	return result
+}
+
+func validateTags(tags map[string]string) validators.ValidationError {
+	var result validators.ValidationError
+	for name, value := range tags {
 		if value == "" {
-			result.AddViolationAt(validators.RootedAt("tags").Key(name), `tag value cannot be empty`)
+			result.AddViolationAt(validators.RootedAt("tags").Key(name), `value cannot be empty`)
+		}
+		if !nameCharacterSet.MatchString(name) {
+			result.AddViolationAt(validators.RootedAt("tags").Key(name), `key must consist of alphanumeric characters, dots, dashes and underscores`)
+		}
+		if !nameCharacterSet.MatchString(value) {
```

Comment:
You also allow `*` here. I think dataplane value tag should have separate regex without `*`

Reply:
No, I believe I don't: 
```
var nameCharacterSet = regexp.MustCompile(^[a-zA-Z0-9\.\-_:]*$)
var selectorCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*|\*)$`)
```
I was thinking 2 sets are enough, but maybe I should reconsider naming

Reply:
You are right, sorry. I think it makes sense to have
```
var tagNameCharacterSet = regexp.MustCompile(^[a-zA-Z0-9\.\-_:]*$)
var selectorCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*|\*)$`)
var tagValueCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*)$`)
```